app.js

```diff
@@ -7,67 +7,65 @@ const middleEl = document.getElementById('middle');
 const bottomEl = document.getElementById('bottom');
 const reportEl = document.getElementById('report');
 const catchphrasesEl = document.getElementById('catchphrases');
-const catchphraseInput = document.getElementById('catchphrase-input');
+const catchphraseInputEl = document.getElementById('catchphrase-input');
 const catchphraseButton = document.getElementById('catchphrase-button');
 
 // set state for how many times the user changes the head, middle, and bottom
+let headCount = 0;
+let middleCount = 0;
+let bottomCount = 0;
 // set state for all of the character's catchphrases
-
+let catchphrases = [];
+// Events
 headDropdown.addEventListener('change', () => {
-    // get the value of the head dropdown
-
-    // increment the head change count state
-    
-    // update the dom for the head (use style.backgroundImage on the headEl div instead of trying to set the .src -- it's NOT an img tag!)
-
-    // update the stats to show the new count (call displayStats() to do this work)
+    const userClicked = headEl.value;
+    headCount++;
+    displayStats();
 });
-
-
 middleDropdown.addEventListener('change', () => {
     // get the value of the middle dropdown
-
+    const userClicked = middleEl.value;
     // increment the middle change count state
-    
+    middleCount++;
     // update the dom for the middle (NOTE: use style.backgroundImage on the middleEl div instead of trying to set the .src -- it's NOT an img tag!)
-
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
-
 bottomDropdown.addEventListener('change', () => {
     // get the value of the bottom dropdown
-
+    const userClicked = bottomEl.value;
     // increment the bottom change count state
-    
+    bottomCount++;
     // update the dom for the bottom (NOTE use style.backgroundImage on the bottomEl div instead of trying to set the .src -- it's NOT an img tag!)
-
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
-catchphraseButton.addEventListener('click', () => {
+catchphrasesEl.addEventListener('click', () => {
     // get the value of the catchphrase input
-    
+    const inputPhrase = catchphraseInputEl.value;
     // push the new catchphrase to the catchphrase array in state
-
+    catchphrases.push(catchphraseInputEl.value);
     // clear out the form input's value so it's empty to the user
-   
+    catchphraseInputEl.value = '';
     // update the dom to show the new catchphrases (refactor to/call displayCatchphrases to do this work)
-
+    displayCatchphrases();
 });
 
+function makeURL(location, id) {
```

Comment:
cool! great use of a pure function here. looks like you just need to call this in your head, middle, and bottom event listeners and pass some arguments.